src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala

```diff
@@ -189,22 +295,90 @@ trait GraphDAO {
+          throw new RawlsException(s"required property [${paramName}] of class [${classT.fullName}] does not have a value in ${vertex}")
+        }
+
+        case optionSymbol if optionSymbol <:< typeOf[Option[_]] => vertexProperty
+
+        case workspaceNameSymbol if workspaceNameSymbol =:= typeOf[WorkspaceName] => workspaceName.getOrElse(throw new RawlsException("workspace name not supplied but is required"))
+
+        case _ => vertexProperty.getOrElse(throw new RawlsException(s"required property [${paramName}] of class [${classT.fullName}] does not have a value in ${vertex}"))
       }
+    }
+
+    try {
+      ctorMirror(parameters: _*).asInstanceOf[T]
+    } catch {
+      case e: Throwable =>
+        println(ctorMirror(parameters: _*))
+        throw e
+    }
+  }
+
+  private def deserializeDomainObjects[T: TypeTag](vertex: Vertex, edgeLabel: String, workspaceName: Option[WorkspaceName]): Seq[T] = {
+    vertex.getVertices(Direction.OUT, edgeLabel).map { v =>
+      loadFromVertex[T](v, workspaceName)
+    }.toSeq
+  }
 
-      if (paramSymbol.typeSignature <:< typeOf[Option[_]]) {
-        prop
-      } else {
-        prop.getOrElse(throw new RawlsException(s"required property [${paramName}] of class [${classT.fullName}] does not have a value in ${vertexProperties}"))
+  private def deserializeMap(vertex: Vertex): Map[String, Attribute] = {
+    vertex.asInstanceOf[OrientVertex].getRecord.setAllowChainedAccess(false)
+
+    val scalarProps: Map[String, Attribute] = getVertexProperties(vertex).map(entry => entry._1 -> AttributeConversions.propertyToAttribute(entry._2))
+
+    val refProps = vertex.getEdges(Direction.OUT).map { edge =>
+      val subV = edge.getVertex(Direction.IN)
+      getVertexClass(subV) match {
+        case VertexSchema.Map =>
+          val attributes: Seq[Attribute] = deserializeSeq(subV)
+          val typedList = attributes.headOption match {
+            case Some(v: AttributeValue) => AttributeValueList(attributes.map(_.asInstanceOf[AttributeValue]))
+            case Some(r: AttributeReference) => AttributeReferenceList(attributes.map(_.asInstanceOf[AttributeReference]))
+            case None => AttributeEmptyList
+          }
+          edge.getLabel -> typedList
+        case VertexSchema.Entity => edge.getLabel -> AttributeReference(subV.getProperty("entityType"), subV.getProperty("name"))
+        case _ => throw new RawlsException("unexpected vertex class: " + getVertexClass(subV))
       }
     }
 
-    ctorMirror(parameters: _*).asInstanceOf[T]
+    scalarProps ++ refProps
+  }
+
+  private def deserializeSeq(subV: Vertex): Seq[Attribute] = {
+    // Seqs are stored as maps with the index as key
+    // for some reason, orient converts our numeric edge labels to negatives so use Math.abs
+    SortedMap(deserializeMap(subV).map(entry => Math.abs(entry._1.toInt) -> entry._2).toSeq: _*).values.toSeq
```

Comment:
is this a better solution than for instance "index-0" and stripping off the number from that?